_posts/2020-03-30-debugging-unit-tests-with-honeycomb.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: post
+title: "Debugging unit-tests with Honeycomb"
+author: dev_el_ops
+categories:
+  - module
+  - development
+tags:
+  - modules
+  - CI
+  - debugging
+  - testing
+  - honeycomb
+---
+
+Since [last week]({% link _posts/2020-03-27-status-update.md %}#litmus-progress) we're collecting unit-test results and acceptance test runs in [honeycomb](https://honeycomb.io/), a tool for introspecting and interrogating production systems. In our team's case, CI is the "production" environment we manage. Today Daniel pointed out that there was a curious issue with one of our test suites:
+
+```ruby
+NoMethodError:
+undefined method `message_lines' for #<RSpec::Core::Notifications::SkippedExampleNotification:0x00007f8407ac3018>
+# /Users/danielcarabas/.rvm/gems/ruby-2.6.3/gems/rspec_honeycomb_formatter-0.2.0/lib/rspec_honeycomb_formatter.rb:95:in `example_pending'
+```
+
+That clearly rings a bell, but a trivial try at reproducing this using `pending` brought up a weird issue:
+
+```ruby
+From: /home/david/git/rspec_honeycomb_formatter/lib/rspec_honeycomb_formatter.rb @ line 99 RSpecHoneycombFormatter#example_pending:
+
+     95: def example_pending(notification)
+     96:   @example_span.add_field('rspec.result', 'pending')
+     97:   @example_span.add_field('name', notification.example.description)
+     98:   @example_span.add_field('rspec.description', notification.example.description)
+ =>  99:   require'pry';binding.pry
+    100:   @example_span.add_field('rspec.message', strip_ansi(notification.message_lines.join("\n")))
+    101:   @example_span.add_field('rspec.backtrace', notification.formatted_backtrace.join("\n"))
+    102:   @example_span.send
+    103: end
+
+[1] pry(#<RSpecHoneycombFormatter>)> notification.class
+=> RSpec::Core::Notifications::PendingExampleFailedAsExpectedNotification
+[2] pry(#<RSpecHoneycombFormatter>)>
+```
+
+The class here is not the same as the initial report is complaining about.
+And this does not fail.
+`PendingExampleFailedAsExpectedNotification` does have `message_lines`.
+Since the original report didn't any more information attached (and for the sake of the pacing of this blog post) I went to honeycomb to see if other modules were exhibiting the same problem.
```

Comment:
```suggestion
Since the original report didn't have any more information attached (and for the sake of the pacing of this blog post) I went to honeycomb to see if other modules were exhibiting the same problem.
```